src/main/kotlin/com/jetbrains/snakecharm/lang/psi/types/SmkPepConfigType.kt

```diff
@@ -0,0 +1,87 @@
+package com.jetbrains.snakecharm.lang.psi.types
+
+import com.intellij.codeInsight.lookup.LookupElement
+import com.intellij.codeInsight.lookup.LookupElementBuilder
+import com.intellij.openapi.roots.ProjectRootManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiManager
+import com.intellij.psi.impl.source.tree.LeafPsiElement
+import com.intellij.util.ProcessingContext
+import com.jetbrains.python.psi.AccessDirection
+import com.jetbrains.python.psi.PyExpression
+import com.jetbrains.python.psi.impl.ResolveResultList
+import com.jetbrains.python.psi.resolve.PyResolveContext
+import com.jetbrains.python.psi.resolve.RatedResolveResult
+import com.jetbrains.python.psi.types.PyType
+import com.jetbrains.snakecharm.codeInsight.resolve.SmkResolveUtil
+import com.jetbrains.snakecharm.lang.psi.SmkFile
+import com.jetbrains.snakecharm.lang.psi.impl.SmkPsiUtil
+import org.jetbrains.yaml.psi.YAMLFile
+
+class SmkPepConfigType(private val smkFile: SmkFile) : PyType {
+    override fun resolveMember(
+        name: String,
+        location: PyExpression?,
+        direction: AccessDirection,
+        resolveContext: PyResolveContext
+    ): List<RatedResolveResult> {
+        if (!SmkPsiUtil.isInsideSnakemakeOrSmkSLFile(location)) {
+            return emptyList()
+        }
+        val resolveResult = ResolveResultList()
+        sectionArgs
+            .filter { it.text == name }
+            .forEach {
+                resolveResult.poke(it, SmkResolveUtil.RATE_NORMAL)
+            }
+        return resolveResult
+    }
+
+    override fun getCompletionVariants(
+        completionPrefix: String?,
+        location: PsiElement?,
+        context: ProcessingContext?
+    ): Array<LookupElement> {
+        return sectionArgs.map {
+            LookupElementBuilder
+                .create(it.text)
+        }.toTypedArray()
+
+    }
+
+    private val sectionArgs = getYamlKeys(getYamlFile(smkFile))
+
+    override fun getName(): String? {
+        return "pep.config"
+    }
+
+    override fun isBuiltin(): Boolean {
+        return false
+    }
+
+    override fun assertValid(message: String?) {
+    }
+
+    companion object {
+        private fun getYamlKeys(yamlFile: PsiFile): List<PsiElement> {
+            val ymlFile = yamlFile as YAMLFile
+            val completionList = mutableListOf<PsiElement>()
+            ymlFile.documents.forEach { document ->
+                document.topLevelValue!!.children.forEach { child ->
+                    completionList.add(child.firstChild)
+                }
+            }
+            return completionList
+        }
+
+        private fun getYamlFile(smkFile: SmkFile): PsiFile {
+            val psiElement = smkFile.findPepfile()!!.getSectionKeywordNode() as LeafPsiElement
```

Comment:
The whole method 1) is using low level PSI access and parser impl details 2) duplicates some code from SmkPepfileReference reference 3) doesn't work properly if reference uses single quoting

Re-implement this using high-level PSI access (e.g. without LeafPsiElement, nextSibling, lastChild, text methods). Use the fact, that if everyting is ok, you already have `SmkPepfileReference` reference here and it is know how to resolve it's target, no need to copy resolve implementation here. Before commiting this change please show me new impl in Slack.